core/templates/dev/head/services/StateTopAnswersStatsService.ts

```diff
@@ -43,15 +41,24 @@ oppia.factory('StateTopAnswersStatsService', [
      * @param {string} stateName
      */
     var refreshAddressedInfo = function(stateName) {
+      if (!stateTopAnswersStatsCache.hasOwnProperty(stateName)) {
```

Comment:
Seems like an error if this gets called in that case ...?

Reply:
Not necessarily, it could be helpful to have when a creator introduces a new state with the same one of a previously deleted one.

Reply:
But wouldn't that state have been deleted from the cache a while back?

Reply:
(Also, if we don't expect this to happen in practice, then maybe we should log an error if it does so we can debug.)

Reply:
Right, that's why we return early if it doesn't exist here. The flow this respects is:

1. Got stats from state 'A' from backend. (cache contains 'A')
2. Creator deletes 'A'. (cache deletes 'A')
3. Creator creates a new state with name 'A' later on (cache does nothing, because no 'A' exists)

Reply:
Also, it's not an error for a state to have no stats. New cards, for example, fit this scenario.

Reply:
Oh yep, I agree with that. Let me make my question a little clearer: in the case where stateName is not in stateTopAnswersStatsCache, shouldn't refreshAddressedInfo not be called at all?

Even in the flow you mention, presumably when a new state is created (or rather when a new interaction is created for that state), the cache gets initialized with that state name plus an empty array.

Reply:
Mk, after taking a look at the backend code and writing tests, I agree this makes more sense as an error.